chapters/packages.tex

```diff
@@ -321,3 +321,146 @@ \section{External Resources}\label{external-resources}
 \filename{modelica://Modelica/Mechanics/C.jpg} is illegal.
 The references \filename{modelica://Modelica.Mechanics/C.jpg} and \filename{modelica://Modelica/C.jpg} must also refer to two distinct resources.
 \end{example}
+
+\section{Multilingual Descriptions}\label{multilingual-descriptions}
+
+\begin{nonnormative}
+Descriptive texts in a model or library are usually formulated in English.
+This section describes how a tool can present the library in another language.
+Translated Modelica text is provided by external files, so that no modification of the original Modelica text is required.
+\end{nonnormative}
+
+The texts in following Modelica constructs should be translated:
+\begin{itemize}
+\item description strings of component declarations and classes
+\item strings in the following annotations:
+\begin{itemize}
+\item \lstinline!Text.string!, \lstinline!Text.textString!
+\item \lstinline!missingInnerMessage!, \lstinline!obsolete!, \lstinline!unassignedMessage!
+\item \lstinline!Dialog.group!, \lstinline!Dialog.tab!
+\item \lstinline!Dialog.loadSelector.caption!, \lstinline!Dialog.loadSelector.filter!, \lstinline!Dialog.saveSelector.caption!, \lstinline!Dialog.saveSelector.filter!
+\item \lstinline!Documentation.info!, \lstinline!Documentation.revisions!
+\item \lstinline!Figure.title!, \lstinline!Plot.title!, \lstinline!Curve.legend!
+\end{itemize}
+\end{itemize}
+
+C-style and line comments are not translated.
```

Comment:
```suggestion
C-style and line comments are not translated.
Only literal strings and concatenations of literals strings are translated.
In order to have parameter values as part of the texts the special substitution syntax is preferable (see \cref{text} and \cref{variable-replacements}) and translations should preserve those substrings.
```

Reply:
I just realized we have established a terminology for the comments:
```suggestion
Comments (delimited as well as rest-of-line) are not translated.
Only literal strings and concatenations of literals strings are translated.
In order to have parameter values as part of the texts the special substitution syntax is preferable (see \cref{text} and \cref{variable-replacements}) and translations should preserve those substrings.
```

Reply:
The proposed formulation isn't clear enough; we need a precise definition of what is expected to go into the template files.
```suggestion
Comments (delimited as well as rest-of-line) are not translated.
Where translation applies, the strings to be translated are the maximum subexpressions consisting of one or more concatenated string string literals.

\begin{example}
In the following annotation, the strings to be translated are \emph{Hello, AAA} and \emph{BBB}:
\begin{lstlisting}[language=modelica]
annotation(…, Text(string = "Hello, " + "AAA" + getInstanceName() + "BBB"), …);
\end{lstlisting}
\end{example}

\begin{nonnormative}
In order to have parameter values as part of the texts, the special substitution syntax is preferable (see \cref{text} and \cref{variable-replacements}) and translations should preserve those substrings.
\end{nonnormative}
```

Reply:
So, if I understnd it rigtht. in the example the substrings are translated separately. I would agree with such solution.

Reply:
Yes, that's correct.  (The alternative design would have been to say that there is nothing in this `Text.string` that can be translated, since only constructs _entirely_ given by one or more concatenated strings would be subject to translation.)

Reply:
The alternative design was the intended one, as this would otherwise attempt to translate _parts of a sentence_ separately which is simply not good, which was the reason to link to `%par` (and `%name`) which are better ways of handling this. I also strongly believe that examples of translation shall have actual English texts and not faked ones - as that will make such issues more obvious.

Reply:
OK, I'll alter the example:
```suggestion
Comments (delimited as well as rest-of-line) are not translated.
Where translation applies, the strings to be translated are the maximum subexpressions consisting of one or more concatenated string string literals.

\begin{example}
Let \lstinline!stateSelectName! be a helper function that selects a string based on a \lstinline!StateSelect! value:
\begin{lstlisting}[language=modelica]
function stateSelectName
  input StateSelect stateSelect;
  input String[5] names "Name of StateSelect enumerators, in the order never, avoid, default, prefer, always";
  output String name;
  $\ldots$
end stateSelectName;
\end{lstlisting}
Consider the following \lstinline!Text.string! (which is a construct where translation applies) used to show the state selection setting in an icon:
\begin{lstlisting}[language=modelica]
annotation(…, Text(string =
    "State " +
    "selection: " +
    stateSelectName(stateSelect, {"never", "avoid", "default", "prefer", "always"}),
 …);
\end{lstlisting}
The strings to be translated are:
\begin{itemize}
\item ``State selection: ''
\item ``never''
\item ``avoid''
\item ``default''
\item ``prefer''
\item ``always''
\end{itemize}
\end{example}

\begin{nonnormative}
In order to have parameter values as part of the texts, the special substitution syntax is preferable (see \cref{text} and \cref{variable-replacements}) and translations should preserve those substrings.
\end{nonnormative}
```

As the use of format strings is a very good idea for shaping meaningful strings for translation, I also see a clear need to support constructs like this:
```
text = someKindOfSprintf("Start (%1): %startParam", (if startFixed then "fixed" else "non-fixed")))
```
That is, the main string to be translated appears as an argument to some string formatting function.  (The formatting function could be both a constant-evaluable user-defined function, or something provided by the language in the future.)

Reply:
To me this example is too long and tangential compared to the topic, and maximum subexpressions doesn't seem to correspond to the decision. 

Additionally I believe that at least the preservation of `%p` should be normative - and even if the rest of the sentence could be non-normative we would then sort of have a normative text depending on the non-normative part which seems odd.

Reply:
> To me this example is too long and tangential compared to the topic, and maximum subexpressions doesn't seem to correspond to the decision.

I'm not sure which decision you refer to, but I agree that the example is too long.  It just ended up longer because the originally suggested example was considered a fake.  I'll start over by suggesting a shorter example again.